static/index.html

```diff
@@ -66,35 +64,39 @@
 		}
 
 		function clockTick() {
-			var d = new Date();
-			var hour = ((d.getHours() % 12) || 12);
+			let d = new Date();
+			let hour = ((d.getHours() % 12) || 12);
 			//hour = (hour < 10 ? "&nbsp;" : "") + hour; // Pad hour
-			var colon = d.getSeconds() % 2 ? ":" : " ";
-			var min = (d.getMinutes() < 10 ? "0" : "") + d.getMinutes()
-			var amOrPm = d.getHours() >= 12 ? "pm" : "am";
+			let colon = d.getSeconds() % 2 ? ":" : " ";
+			let min = (d.getMinutes() < 10 ? "0" : "") + d.getMinutes()
+			let amOrPm = d.getHours() >= 12 ? "pm" : "am";
 			document.getElementById("clock").innerHTML = hour + colon + min + amOrPm;
 		}
 		
 		function readSensor(endpoint, callback) {
-			var xhttp = new XMLHttpRequest();
+			let xhttp = new XMLHttpRequest();
 			xhttp.onreadystatechange = function() {
 				if (this.readyState == 4 && this.status == 200) {
-					callback(this.responseText);
+					callback(this.response);
 				} else {
 					// Display error to driver?
 				}
 			};
 			xhttp.open("GET", endpoint + "?" + Date.now(), true); // Date.now() prevents caching
 			xhttp.send();
 		}
-		
-		function updateSpeed() {
-			readSensor("/get-speed", speed => speedometer.set(speed));
-		}
-		function updateBatteryPercent() {
-			readSensor("/get-battery-percent", percent => document.getElementById("battery-percent").innerText = percent);
+
+		function populateData(data) {
+			console.log(data)
+			speedometer.set(0) //data["speed"])
+			document.getElementById("battery-percent").innerText = data["soc"]
+			document.getElementById("temperature").innerHTML = data["temp"] + "&deg;C"
```

Comment:
Tiny detail: It would probably make more sense to move the &deg;C outside of the span than to update with innerHTML and replace the unit every time

Reply:
Makes sense